wrappers/javascript/shared/src/indyVdr/IndyVdrRequest.ts

```diff
@@ -17,11 +17,15 @@ export type RequestSetEndorserOptions = {
   endorser: string
 }
 
-export type RequestResponseType<Request extends IndyVdrRequest> = NonNullable<Request['__responseType__']>
+export type RequestResponseType<Request> = Request extends IndyVdrRequest<infer ResponseType> ? ResponseType : never
```

Comment:
Shouldn't it still be nonnullable?

Reply:
No because we don't use a property anymore to infer the type, we infer it directly from the generic (using the `infer ResponseType`)